File: src/main/java/ha02/a3/Inbox.java
package ha02.a3;

import java.util.ArrayList;

public class Inbox {
    ArrayList<Email> emails = new ArrayList<>();

    public Inbox(){}

    public void addMail(Email e){
        emails.add(e);
    }

    public void removeMail(Email e){
        emails.remove(e);
    }

    public ArrayList<String> printAllEmails(){
        ArrayList<String> printedMails = new ArrayList<>();

        for (Email email : emails) {
            printedMails.add(
                    (email.isRead() ? "Read" : "Unread") + " | " +
                    email.getSubject() + " | " +
                    email.getSenders_address() + " | " +
                    email.getDatetime());
        }

        return printedMails;
    }

    public String open(int index){
        try{
            if(index < emails.size()){
                return emails.get(index).print();
            }
        } catch(IndexOutOfBoundsException e) {
            throw new IndexOutOfBoundsException();
        }
        return null;
    }

    public int countUnread(){
        int i = 0;

        for(Email email : emails){
            if(!email.isRead()){
                i++;
            }
        }
        return i;
    }
}

File: src/main/java/ha10/a3/Inbox.py
class Inbox:
    def __init__(self):
        self._mails = []

    def add_mail(self, mail):
        self._mails.append(mail)

    def remove_mail(self, mail):
        if mail in self._mails:
            self._mails.remove(mail)

    def open(self, index):
        try:
            return self._mails[index].printMail()
        except IndexError:
            raise IndexError("Index out of bound")

    def count_unread(self):
        return sum(1 for mail in self._mails if not mail.is_read())

    def print_all_emails(self):
        printed_mails = []

        for mail in self._mails:
            status = "Read" if mail.is_read() else "Unread"
            line = f"{status} | {mail.get_subject()} | {mail.get_senders_address()} | {mail.get_datetime()}"
            printed_mails.append(line)

        return printed_mails

